aurora-solidity-sdk/src/AuroraSdk.sol

```diff
@@ -0,0 +1,217 @@
+            if (status == PromiseResultStatus.Successful) {
+                borsh.skipBytes();
+            }
+        }
+
+        result.status = PromiseResultStatus(borsh.decodeU8());
+        if (result.status == PromiseResultStatus.Successful) {
+            result.output = borsh.decodeBytes();
+        }
+    }
+
+    /// Get the NEAR account id of the current contract. It is the account id of Aurora engine.
+    function currentAccountId() public returns (string memory) {
+        (bool success, bytes memory returnData) = CURRENT_ACCOUNT_ID_PRECOMPILE.call("");
+        require(success);
+        return string(returnData);
+    }
+
+    /// Get the NEAR account id of the predecessor contract.
+    function predecessorAccountId() public returns (string memory) {
+        (bool success, bytes memory returnData) = PREDECESSOR_ACCOUNT_ID_PRECOMPILE.call("");
+        require(success);
+        return string(returnData);
+    }
+
+    /// Crease a base promise. This is not immediately schedule for execution
+    /// until transact is called. It can be combined with other promises using
+    /// `then` combinator.
+    ///
+    /// Input is not checekd during promise creation. If it is invalid, the
+    /// transaction will be scheduled either way, but it will fail during execution.
+    function call(
+        NEAR storage near,
+        string memory targetAccountId,
+        string memory method,
+        bytes memory args,
+        uint128 nearBalance,
+        uint64 nearGas
+    ) public returns (PromiseCreateArgs memory) {
+        /// Need to capture nearBalance before we modify it so that we don't
+        /// double-charge the user for their initialization cost.
+        PromiseCreateArgs memory promise_args = PromiseCreateArgs(targetAccountId, method, args, nearBalance, nearGas);
+
+        if (!near.initialized) {
+            /// If the contract needs to be initialized, we need to attach
+            /// 2 NEAR (= 2 * 10^24 yoctoNEAR) to the promise.
+            nearBalance += 2_000_000_000_000_000_000_000_000;
+            near.initialized = true;
+        }
+
+        if (nearBalance > 0) {
+            near.wNEAR.transferFrom(msg.sender, address(this), uint256(nearBalance));
+        }
+
+        return promise_args;
+    }
+
+    /// Similar to `call`. It is a wrapper that simplifies the creation of a promise
+    /// to a controct inside `Aurora`.
+    function auroraCall(NEAR storage near, address target, bytes memory args, uint128 nearBalance, uint64 nearGas)
+        public
+        returns (PromiseCreateArgs memory)
+    {
+        return call(
+            near,
+            currentAccountId(),
+            "call",
+            abi.encodePacked(uint8(0), target, uint256(0), args.encode()),
+            nearBalance,
+            nearGas
+        );
+    }
+
+    /// Schedule a base promise to be executed on NEAR. After this function is called
+    /// the promise should not be used anymore.
```

Comment:
It cannot be used to obtain the result because the result is always only obtainable from the `promiseResult` function inside a callback. I've added a brief mention of this to the comment in https://github.com/aurora-is-near/aurora-contracts-sdk/pull/4/commits/33dcfbc3f27ea2e54b8070cc7e82a617244f0c1b